SmartContracts.NET/ContractsCore/ContractRegistry.cs

```diff
@@ -47,30 +49,70 @@ public void RegisterContract(Contract contract)
 			}
 
 			this.addressesToContracts[address] = contract;
-			contract.Send += this.HandleSendActionEvent;
-			contract.Forward += this.HandleForwardActionEvent;
+			contract.Send += (x, actionArgs) =>
```

Comment:
Can't we change the signature of Handle__Event and not use a lambda?

Reply:
this is the only way we are sure that the Target is contract's address. Otherwise we are not sure what value will be passed when the event is invoked.